apps/meteor/client/startup/banners.ts

```diff
@@ -15,14 +15,20 @@ const fetchInitialBanners = async (): Promise<void> => {
 
 	for (const banner of response.banners) {
 		if (banner._id === 'device-management') {
-			setTimeout(() => {
-				imperativeModal.open({
-					component: DeviceManagementFeatureModal,
-					props: {
-						close: imperativeModal.close,
-					},
-				});
-			}, 2000);
+			Tracker.autorun((computation) => {
+				const user = Meteor.user();
+				if (user?.username) {
+					setTimeout(() => {
+						imperativeModal.open({
+							component: DeviceManagementFeatureModal,
+							props: {
+								close: imperativeModal.close,
+							},
+						});
+					}, 2000);
+					computation.stop();
+				}
+			});
```

Comment:
would you mind mind return if username is empty?

Reply:
Done!